digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/ProjectDeltaCalculator.scala

```diff
@@ -83,8 +95,24 @@ object ProjectDeltaCalculator {
     ).toSeq
   }
 
+  def partition(roadAddresses: Seq[RoadAddress], projectLinks : Seq[ProjectLink]): (Seq[RoadAddressSection],Seq[RoadAddressSection]) = {
+    val groupedAddresses = roadAddresses.sortBy(_.startAddrMValue).groupBy(ra => (ra.roadNumber, ra.roadPartNumber, ra.track))
+    val addressesGroups = groupedAddresses.mapValues(v => combine(v.sortBy(_.startAddrMValue))).values.flatten.map(ra =>
```

Comment:
Will fail if there are multiple, possibly gapped transfer sections on road.

Instead, map by link id and group then.
RoadAddress.linkId == ProjectLink.linkId on transfer.

Reply:
Could also return a Map[RoadAddressSection, RoadAddressSection] which would give the correct mapping row immediately